plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -435,23 +435,21 @@ codeActionProvider ideState _pluginId (CodeActionParams _ _ documentId _ context
     diags = context ^. LSP.diagnostics
 
 resolveProvider :: Recorder (WithPriority Log) -> PluginMethodHandler IdeState Method_CodeActionResolve
-resolveProvider recorder ideState _pluginId ca@CodeAction {_data_ = Just data_} = pluginResponse $ do
-  case fromJSON data_ of
-    (Success (AA verTxtDocId@(VersionedTextDocumentIdentifier uri _))) -> do
-        file <-  getNormalizedFilePath uri
-        edit <- ExceptT $ liftIO $ applyHint recorder ideState file Nothing verTxtDocId
-        pure $ ca & LSP.edit ?~ edit
-    (Success (AO verTxtDocId@(VersionedTextDocumentIdentifier uri _) pos hintTitle)) -> do
-        let oneHint = OneHint pos hintTitle
-        file <-  getNormalizedFilePath uri
-        edit <- ExceptT $ liftIO $ applyHint recorder ideState file (Just oneHint) verTxtDocId
+resolveProvider recorder ideState _
+  ca@CodeAction {_data_ = Just (fromJSON -> (Success (ApplyHint verTxtDocId oneHint)))} = pluginResponse $ do
+        file <-  getNormalizedFilePath (verTxtDocId ^. LSP.uri)
```

Comment:
The reason why we can't is because we need the uri to do anything useful, and that's encoded in the data. I'm gonna try and see if I can model resolve provider infrastructure on what we have for commands. That way hls core can do the json decoding and also store a uri to pass on to the handlers for us to use right away.

Reply:
Ah right, I see. Yeah, maybe that's natural. But then again maybe we should leave this stuff up to plugins. Something like the URI you _definitely_ always need though.